rclpy/test/__init__.py

```diff
@@ -12,18 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import importlib
+import os
 import sys
 
 assert 'rclpy' not in sys.modules, 'rclpy should not have been imported before running tests'
 
-# this will make the extensions load from the build folder
-import rclpy.impl  # noqa
-import test_rclpy  # noqa
+# this will load rclpy from the build folder
+sys.path.insert(0, os.environ.get('RCLPY_TEST_LIBRARY_DIR'))
```

Comment:
@ivanpauno using that envvar is so shady... there has to be a natural way to do this. I wonder what `setuptools` does under the hood.